### ISSUE TITLE
Fix setting `onFocus` callback for `SemanticsAction.focus` on iOS.

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter#149838]
## Description

In trying to land https://github.com/flutter/flutter/pull/142942, the iOS platform view test failed, so it was [reverted](https://github.com/flutter/flutter/pull/149741).

Jen (@jmagman ), Yegor (@yjbanov) and I have tried debugging this and reached this level of understanding of the bug.

Jen explained in this comment the details we've discovered so far: https://github.com/flutter/flutter/pull/149741#issuecomment-2151093664

I've reached the limit in the amount of time I can devote to this, so I'm going to just prevent setting the callback on iOS (where it isn't expected to do anything anyhow), and submit it for other platforms, and we can debug it at our leisure.

The behavior is definitely very odd: Setting the `onFocus` callback for the `Semantics` node in the `Focus` widget in the framework causes the _native_ iOS text field in the [platform view UI integration test](https://github.com/flutter/flutter/blob/master/dev/integration_tests/ios_platform_view_tests/ios/PlatformViewUITests/PlatformViewUITests.m) to shift left by 93 pixels so that it's lower left corner is outside of the bounding box for the platform view, consequently making it not possible to scroll it into view when the test tries to tap on it. If the callback isn't set, it doesn't move the text field. I'm going to mention again that it's not a Flutter text field, it's the _native_ iOS text field inside the platform view that is shifting position, just for emphasis.

In no way does this make sense to me, and it should be debugged to find out the root cause, since I suspect it is a deeper bug lurking in the iOS engine, but since iOS isn't expected to send focus events separate from accessibility focus events, the callback wouldn't ever get called anyhow.

For now, I'm filing this issue and attaching it to a TODO in the code.

**Comment by jmagman** [comment]
Tried on https://github.com/flutter/engine/pull/53285/commits/205a24192f67502b26bb52b00a635b7925736a01 (which didn't work), documenting how I did that:

Build the iOS simulator and host engine with the fix (my machine is arm64):
```
$ gh pr checkout 53285
$ et build -c ios_debug_sim_arm64 && et build -c host_debug_unopt_arm64
```
In the framework cherry-pick https://github.com/flutter/flutter/pull/142942:
```
$ git cherry-pick dd700e6
```
Set up the test to run against the local engine with the fix (see https://github.com/flutter/flutter/pull/149741#issuecomment-2151093664 for more details):
```
$ cd dev/integration_tests/ios_platform_view_tests
$ flutter build ios --config-only --local-engine ios_debug_sim_arm64 --local-engine-host host_debug_unopt_arm64
$ open ios/Runner.xcworkspace
```
In Xcode open `PlatformViewUITests` and run `testPlatformViewFocus`.

**Comment by vashworth** [comment]
Closing in favor of https://github.com/flutter/flutter/issues/150030.
